api/envoy/api/v2/core/base.proto

```diff
@@ -95,6 +95,78 @@ message Metadata {
   map<string, google.protobuf.Struct> filter_metadata = 1;
 }
 
+// MetadataMatcher provides a general interface to check if a given value is matched in
+// :ref:`Metadata <envoy_api_msg_core.Metadata>`. It uses `filter` and `path` to retrieve the value
+// from the Metadata and then check if it's matched to one of the specified values.
+//
+// An example use of MetadataMatcher is specifying additional metadata in envoy.filters.http.rbac to
+// enforce access control based on dynamic metadata in a request.
+message MetadataMatcher {
+  // Specifies the value to match. Only primitive value is supported.
+  message Value {
+    // Specifies how to match a value. Only have effect on primitive value.
+    // For non-primitive value, it's always not matched.
+    oneof match_specifier {
+      option (validate.required) = true;
+
+      // If specified, value match will be performed based on null value. It's matched if and only
+      // if the target value is a NullValue and this field is set to true.
+      bool null_match = 1;
+
+      // If specified, value match will be performed based on double value. It's matched if and only
+      // if the target value is a double value and is equal to this field.
+      double number_match = 2;
```

Comment:
floating number is (almost) impossible to do exact match, consider add a range match like `envoy.type.Int64Range`

Reply:
Updated to use a DoubleMatcher.